### DISCUSSION TITLE
Offline viewing of cloned repo?

### DISCUSSION THREAD

**Discussion by rgsiiiya** [willwulfken/MidJourney-Styles-and-Keywords-Reference#26] [Q&A]
I want to clone the repo locally so i can have more performant browsing through the docs (I have slow internet). 

Cloning the repo is easy enough, but what is the best .md file viewer to use?
Markdown app does not fully support github md syntax.

Suggestions?

Thanks!
Robert

**Comment by rgsiiiya** [comment]
It's not really what i'm looking for, but the "Markdown Preview Github Styling" VSCode extension does do a good job rendering the .md file in the VSCode preview pane.  but you still have to look at the .md file.  Would prefer a direct MD browser.

**Reply by rgsiiiya** [reply]
Looking at the .md files, they all use absolute img src URLs.  So even when you clone the repo, and use a viewer, it still pulls all the images down from github.

Does github support relative URLs in md files?

**Reply by willwulfken** [reply]
You can open find & replace on your text editor of choice and replace "https://github.com/willwulfken/MidJourney-Styles-and-Keywords-Reference/blob/main" with nothing. That will convert all URLs to relative

**Reply by willwulfken** [reply]
All URLs pointing to this repo have been changed to relative

**Reply by rgsiiiya** [reply]
THANKS! I'll refresh my repo and give it a try!

**Comment by rememple** [comment]
I successfully downloaded all the folder and image, I can see the img URLs is relative, but I still cannot preview the pages. How can I preview the pages in my local computer? Is there so blog tools could do that? please, thank you!

**Reply by colaxiaoyan** [reply]
I'm a designer and I don't know code, but it looks like there might be a parent folder name "MidJourney-Styles-and-Keywords-Reference-main" missing in all the links, maybe add this file name at the beginning of the path The folder name will be displayed normally.
Remember to reply me if you solve it, thanks bro(Tears of Joy).

**Reply by willwulfken** [reply]
So basically yes, they are relative URLs. If you add the github link to the beginning it'll link to github. The difficulty with viewing it locally through markdown is that you may need to add "..\" for each time you need to go back a folder to get the the main folder (from the page of the markdown file)

I'm trying to decide which way I want to combat this, but lately I haven't had much time to work on that unfortunately. It is planned however.

**Reply by colaxiaoyan** [reply]
> So basically yes, they are relative URLs. If you add the github link to the beginning it'll link to github. The difficulty with viewing it locally through markdown is that you may need to add ".." for each time you need to go back a folder to get the the main folder (from the page of the markdown file)
> 
> I'm trying to decide which way I want to combat this, but lately I haven't had much time to work on that unfortunately. It is planned however.

Add a markdown component to chrome makes it easy to preview. I tried to add a prefolder name to the link to the md in the pages folder, and it works without being placed in the root directory (you can also unzip the secondary folder in the root directory and use it directly). I'm very pleased to be able to deploy locally, thanks for your great sharing!

**Reply by rememple** [reply]
Thanks for you prompt, I try to add a prefix to the folder path, and I install a plugin to my browser, finally I could preview all the pictures in my local PC。best wishes for you! 😄

**Reply by colaxiaoyan** [reply]
> Thanks for you prompt, I try to add a prefix to the folder path, and I install a plugin to my browser, finally I could preview all the pictures in my local PC。best wishes for you! 😄

It's my pleasure 😃
